src/RazorSdk/SourceGenerators/RazorSourceGenerator.cs

```diff
@@ -20,114 +26,143 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
             var razorSourceGeneratorOptions = razorSourceGeneratorOptionsWithDiagnostics.ReportDiagnostics(context);
 
             var sourceItemsWithDiagnostics = context.AdditionalTextsProvider
+                .Where(static (file) => file.Path.EndsWith(".razor") || file.Path.EndsWith(".cshtml"))
                 .Combine(context.AnalyzerConfigOptionsProvider)
-                .Where((pair) => pair.Item1.Path.EndsWith(".razor") || pair.Item1.Path.EndsWith(".cshtml"))
                 .Select(ComputeProjectItems);
 
-            var sourceItems = sourceItemsWithDiagnostics.ReportDiagnostics(context);
+            var sourceItems = sourceItemsWithDiagnostics
+                .ReportDiagnostics(context);
 
-            var references = context.CompilationProvider
-                .WithLambdaComparer(
-                    (c1, c2) => c1 != null && c2 != null && c1.References != c2.References)
-                .Select((compilation, _) => compilation.References);
+            var importFiles = sourceItems.Where(static file =>
+            {
+                var path = file.FilePath;
+                if (path.EndsWith(".razor"))
+                {
+                    return Path.GetFileNameWithoutExtension(path) == "_Imports";
+                }
+                else if (path.EndsWith(".cshtml"))
+                {
+                    return Path.GetFileNameWithoutExtension(path) == "_ViewImports";
```

Comment:
Are imports files the only scenario where we care about things being in the virtual file system?

If so, making sure that Razor works with global usings (https://github.com/dotnet/aspnetcore/issues/34217#issuecomment-882001435) might help us in removing the virtual filesystem concept all together.

Reply:
We can't remove _Imports / _ViewImports since they also support other directives and there are existing projects that already use it.

Reply:
Yeah, I realized that we can't get rid of this shortly after typing it. Alas....